src/main/java/com/netflix/imflibrary/st2067_2/IMFTrackFileResourceType.java

```diff
@@ -109,4 +109,33 @@ public boolean equivalent(IMFBaseResourceType other)
 
         return  result;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        IMFTrackFileResourceType otherImfTrackFileResourceType = (IMFTrackFileResourceType) o;
+        return equivalent(otherImfTrackFileResourceType);
```

Comment:
It is a good idea to keep fields used in the implementation of equals and hashCode identical to each other. In this case, if the equivalent method implementation goes through some changes then equals and hashcode will not match.